daemon/dtls.c

```diff
@@ -384,6 +384,14 @@ struct dtls_cert *dtls_cert() {
 	return ret;
 }
 
+void dtls_cert_free(void) {
+	rwlock_lock_r(&__dtls_cert_lock);
+	obj_put(__dtls_cert);
+	rwlock_unlock_r(&__dtls_cert_lock);
```

Comment:
This should probably be a write lock, together with setting `__dtls_cert = NULL`, but since we're in the shutdown here anyway, it doesn't matter much.